### PR TITLE
[feature]: Add sampling documentation for java

### PR DESCRIPTION
**Pull request by konstantinoskolios**
Hello :)

I've addressed issue #2548 by adding Java Sampling Documentation, referencing the 'Go Sampling' guide. Could you please review and provide feedback for improvements?

Thanks!

### CODE REVIEW

File: content/en/docs/languages/java/sampling.md
Comment (cartermp) [inline content/en/docs/languages/java/sampling.md:36]: So maybe this is the right place to figure out the guidance, but I think it's a little more complicated than this.

Consider a scenario where a service you don't control is upstream of your service. If that service always samples, then your service will always sample too, regardless of your value set for `traceIdRatioBased`. However, if you remove the `parentBased` sampler, then you can end up with messed up traces if a downstream service doesn't also sample in the exact same way.

There's...probably a whole article on production sampling best practices we should write here.

**Comment by cartermp** [comment]
@SylvanasGr re my comment about production guidance -- we don't need to block on it, since the doc as you've written it is good w.r.t. our incomplete guidance on the topic. If we hash it out here quickly and want to add it to this doc, then we'll do that, but don't take the comment as blocking.

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
@open-telemetry/java-approvers PTAL!

**Review by svrnm — APPROVED** [review]

**Review by trask — COMMENTED** [review]
I think most Java users are using the Java agent, Spring Boot Starter, or SDK autoconfigure module (and these are generally our recommended solutions vs SDK manual configuration).

I wonder how we can make clear that this page doesn't apply to those users to avoid confusion?

**Comment by svrnm** [comment]
> I think most Java users are using the Java agent, Spring Boot Starter, or SDK autoconfigure module (and these are generally our recommended solutions vs SDK manual configuration).
> 
> I wonder how we can make clear that this page doesn't apply to those users to avoid confusion?

good call out. On other pages we have a "Note" at the top of the page, e.g. https://opentelemetry.io/docs/languages/java/exporters/ and https://opentelemetry.io/docs/languages/java/resources/

**Comment by cartermp** [comment]
For this:

> I wonder how we can make clear that this page doesn't apply to those users to avoid confusion?

I would propose having a top level section that focuses on the env vars / agent properties. Then we can still have a section below for controlling sampling in code as an alternative.

**Comment by SylvanasGr** [comment]
I'm unsure about how to proceed with this pull request.

Any guidance would be greatly appreciated! 😄

**Comment by zeitlinger** [comment]
Here's a page that has a working example for configuring samplers for the spring starter: https://opentelemetry.io/docs/languages/java/automatic/spring-boot/#exclude-actuator-endpoints-from-tracing

I'd suggest to make links both ways.

**Comment by zeitlinger** [comment]
> I'm unsure about how to proceed with this pull request.
> 
> Any guidance would be greatly appreciated! 😄

I would go ahead with this PR.

It's true that we can do more, but as long as we have a [note](https://github.com/open-telemetry/opentelemetry.io/pull/4199#issuecomment-2022450169) I'd go forward with this PR.

**Comment by svrnm** [comment]
@open-telemetry/java-approvers PTAL

**Review by breedx-splk — COMMENTED** [review]
I think this is a fine start (tho it looks like it currently needs an update). If we want to merge it and improve from there, I think that's fine. We should also include Fabri's suggestions.

What I might call out as opportunities for future improvements:

* Emphasize that most the defaults suffice for most users, and that most users don't need to customize the sampler.
* Make it clear that if you don't use a parent-based sampler to wrap these other selections, you may get incomplete or broken traces.

Thanks @SylvanasGr !

**Review by breedx-splk — APPROVED** [review]

**Comment by cartermp** [comment]
Merged since it's been a hot minute and this is much better as-is than the current state in production

**Pull request merged**
